Assignment6.Tests/ArrayTests.cs

```diff
@@ -0,0 +1,51 @@
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Assignment6;
+using System.Collections.Generic;
+
+namespace Assignment6.Tests
+{
+    [TestClass]
+    public class ArrayTests
+    {
+        [TestMethod]
+        [ExpectedException(typeof(System.ArgumentOutOfRangeException))]
+        public void ArrayConstructor_BadSize_ThrowsException()
+        {
+            var sut = new Array<int>(-1);
+        }
+
+        [TestMethod]
+        [ExpectedException(typeof(System.InvalidOperationException))]
+        [DataRow(-1)]
+        [DataRow(1)]
+        public void ArrayIndex_OutOfRange_ThrowsInvalidOperationException(int index)
+        {
+            var sut = new Array<int>();
+            var _ = sut[index];

```

Comment:
```suggestion
            _ = sut[index];
```
Discard is always in scope.